runtime/core/decoder/torch_asr_decoder.cc

```diff
@@ -272,6 +277,28 @@ void TorchAsrDecoder::UpdateResult() {
   }
 }
 
+float TorchAsrDecoder::LeftToRightScore(const torch::Tensor& prob,
+                                        const std::vector<int>& hyp,
+                                        const int& eos) {
+    float score = 0.0f;
+    for (size_t j = 0; j < hyp.size(); ++j) {
+      score += prob[j][hyp[j]].item<float>();
+    }
+    score += prob[hyp.size()][eos].item<float>();
+    return score;
+}
+
+float TorchAsrDecoder::RightToLeftScore(const torch::Tensor& r_prob,
```

Comment:
The two functions has the same logic, merge it to one, and just reverse the input hyp outside.